File: api/src/main/java/dev/httpmarco/polocloud/api/groups/ClusterGroup.java
package dev.httpmarco.polocloud.api.groups;

import dev.httpmarco.polocloud.api.Detail;
import dev.httpmarco.polocloud.api.Named;
import dev.httpmarco.polocloud.api.platforms.PlatformGroupDisplay;
import dev.httpmarco.polocloud.api.properties.PropertiesPool;
import dev.httpmarco.polocloud.api.services.ClusterService;

import java.util.List;

public interface ClusterGroup extends Named, Detail {

    String[] nodes();

    String[] templates();

    int maxMemory();

    boolean staticService();

    PlatformGroupDisplay platform();

    int minOnlineServerInstances();

    int maxOnlineServerInstances();

    long serviceCount();

    List<ClusterService> services();

    PropertiesPool properties();

    default boolean fallback() {
        return this instanceof FallbackClusterGroup;
    }

}

File: panel/src/lib/api/entities/ClusterGroup.ts
import { Platform } from '@/lib/api/entities/Platform';

export class ClusterGroup {
  name: string = '';
  platform?: Platform;
  maxMemory: number = 0;
  maxPlayers: number = 0;
  fallback: boolean = false;
  staticService: boolean = false;
  minOnlineServerInstances: number = 0;
  maxOnlineServerInstances: number = 0;

  constructor(data: Partial<ClusterGroup>) {
    Object.assign(this, data);
  }
}
